test/http_server_test.exs

```diff
@@ -1,15 +1,38 @@
 defmodule HTTPServerTest do
   use ExUnit.Case
+  alias HTTPServer.Response
+  alias HTTPServer.Request
+  alias HTTPServer.Router
   doctest HTTPServer
 
-  describe "parse_request/1" do
-    #  TODO: expect an error here instead of trying to handle
-    # test "returns an empty request when and empty message is given" do
-    #   message = ""
-    #   expected = %HTTPServer.Request{method: nil, resource: nil, headers: %{}, body: ""}
+  describe "Request/1" do
+    test "returns properly formatted HTTP GET request without body" do
+      message =
+        "GET /simple_get HTTP/1.1\r\n" <>
+          "Host: 127.0.0.1 4000\r\n" <>
+          "User-Agent: ExampleBrowser/1.0\r\n" <>
+          "Accept: */*\r\n" <>
+          "Content-Type: text/plain\r\n" <>
+          "Content-Length: 9\r\n" <>
+          "\r\n" <>
+          ""
+
+      expected_parsed_request = %Request{
+        method: "GET",
+        path: "/simple_get",
+        resource: "HTTP/1.1",
+        headers: %{
+          "Accept" => "*/*",
+          "Content-Length" => "9",
+          "Content-Type" => "text/plain",
+          "Host" => "127.0.0.1 4000",
+          "User-Agent" => "ExampleBrowser/1.0"
+        },
+        body: ""
+      }
```

Comment:
There is already an integration test to test this specific `/simple_get` endpoint.  To keep from coupling your http_server implementation to the `http_server_spec` tests, I suggest not re-using the endpoint names in the unit tests.  

I bet this breaks if you change `GET /simple_get HTTP/1.1\r\n` to `GET /test_get HTTP/1.1\r\n`.  Try mocking out the declaration of that test endpoint, it will open up a world of missing design here.

Reply:
Ok great, I'll make these updates to the tests!

Reply:
The unit tests have been updated and now live in HTTPServerTest.Router. Let me know if you think I could go a bit further to decouple them from the integration test functionality.